control/controlhttp/server.go

```diff
@@ -75,12 +75,30 @@ func AcceptHTTP(ctx context.Context, w http.ResponseWriter, r *http.Request, pri
 	}
 	conn = netutil.NewDrainBufConn(conn, brw.Reader)
 
-	nc, err := controlbase.Server(ctx, conn, private, init)
+	cwc := newWriteCorkingConn(conn)
+
+	nc, err := controlbase.Server(ctx, cwc, private, init)
 	if err != nil {
 		conn.Close()
 		return nil, fmt.Errorf("noise handshake failed: %w", err)
 	}
 
+	if earlyWrite != nil {
+		if deadline, ok := ctx.Deadline(); ok {
+			if err := conn.SetDeadline(deadline); err != nil {
+				return nil, fmt.Errorf("setting conn deadline: %w", err)
+			}
+			defer conn.SetDeadline(time.Time{})
+			if err := earlyWrite(nc.ProtocolVersion(), cwc); err != nil {
```

Comment:
Does calling earlyWrite need to depend on having a deadline? If so it's worth documenting

Reply:
should this be outside the deadline if check?

Reply:
You are still not calling earlyWrite when there is no `ctx.Deadline`

Reply:
If that is intended, that explicitly needs docs.

Reply:
whoops, good catch